src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -616,6 +665,62 @@ private void LoadCustomExtensions()
             generalProvider.CompleteInitialization(metadataProvider);
         }
 
+        private void LoadCustomExtensions(string extensionsPath)
+        {
+            var existing = AppDomain.CurrentDomain.GetAssemblies();
+
+            foreach (var path in Directory.EnumerateFiles(extensionsPath, "*.dll"))
+            {
+                // We don't want to load and reflect over every dll.
+                // By convention, restrict to based on filenames.
+                var filename = Path.GetFileName(path);
+                if (!filename.ToLowerInvariant().Contains("extension"))
+                {
+                    continue;
+                }
+
+                // Skip if we've already loaded it.  Or if it's a builtin.
+                if (IsExtensionAlreadyLoaded(existing, path))
```

Comment:
This logic is probably not what we want. Can we call `LoadFrom` and compare against existing by identity instead? That will be consistent with the `LoadFrom` behavior and also ensure we don't attempt to load an extension if files are duplicated in different paths.

It's eliminates some of the custom logic we have here.